sandbox/libs/authn/src/main/java/org/opensearch/authn/Permission.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.authn;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class Permission {
+    private final static String PERMISSION_DELIMITER = "\\.";  
+    private final String[] permissionChunks; 
+
+    public Permission(final String permission) {
+        try {
+            this.permissionChunks = permission.split(PERMISSION_DELIMITER);
+        } catch (Exception) {
+            throw new InvalidPermissionName(permission);
+        }
```

Comment:
```suggestion
        this.permissionChunks = splitIntoChunks(permission);
```

Reply:
okay, i was thinking it from reusability side of things. Maybe we can change definition of the constructor and the splitIntoChunks `throws InvalidPermissionException`. Thoughts?